swe_roe/ex2b.c

```diff
@@ -1759,14 +1866,31 @@ PetscErrorCode RHSFunctionForBoundaryEdges(RDyApp app, Vec F, PetscReal *amax_va
     if (cells->is_local[l] && b_ptr[l] == 0) {
       // Perform computation for a boundary edge
 
-      if (cells->is_local[l] && b_ptr[l] == 0) {
-        hr_vec_bnd[ii] = hl_vec_bnd[ii];
+      PetscReal uperp, q, velocity;
+      switch (edges->boundary_edge_types[ii]) {
+        case REFLECTING_WALL:
+          hr_vec_bnd[ii] = hl_vec_bnd[ii];
```

Comment:
Did you guarantee the direction of the boundary edge that on the left of it is internal cell and the right of it is ghost cell?

Reply:
`DMPlex` guarantees the cell left of an is internal and the cell on the right is a ghost.

Reply:
Thanks, it is good to know.